### PR TITLE
Bugfix/reset optimization state and fix inference

### PR DESCRIPTION
**Pull request by mosheman5**
A mix of small bugfixes:
- Don't load optimizers state by default to enable model finetuning/transfer learning. Current state is a silent bug since optim arguments have no effect when loading a checkpoint
- Inference given metadata was broken, corrected the dataloaders to return only audio both on `ClassifierDataset` and `InferenceDataset`
- Removed checkpoint hierarchy from conf, now it's under experiment
- changed the conf files for inference
- changed the conf file for mozambique_2018 dataset + added the relevant csv
- Fixed tests (!)

### CODE REVIEW

**Review by noamzbr — APPROVED** [review]
Looks good, left one question

**Comment by mosheman5** [comment]
> 

where is it? Couldn't find the question

File: soundbay/conf/runs/inference_single_audio_manatees.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # @package _global_
 defaults:
   - data: inference_file_manatees
-  - checkpoint: defaults
+  - checkpoint: inference
```
Comment (amitgalor18) [inline soundbay/conf/runs/inference_single_audio_manatees.yaml:4]: You discarded the "checkpoint" section of the hierarchy, didn't you? Is it still supposed to be here?

**Review by amitgalor18 — APPROVED** [review]
Left a couple of comments, about stuff in the conf files. Though I didn't actually find where the main bugfix is

File: soundbay/conf/data/inference_file.yaml [hunk 2]
```diff
@@ -2,11 +2,9 @@
...
   test_dataset:
     _target_: soundbay.data.InferenceDataset
-    file_path: '../data/file1.wav'
+    file_path: '../tests/assets/data/sample.wav'
     data_sample_rate: ${data.data_sample_rate}
     sample_rate: ${data.sample_rate}
   batch_size: 64
   num_workers: 2
-  data_sample_rate: 44100
-save_raven: True
```
[the remark this answers is not in the document]
Comment (mosheman5) [reply soundbay/conf/data/inference_file.yaml:11]: I moved them to the experiment hierarchy. It's false by default but can be easily changed by cmdline argument when running inference

**Pull request merged**